Add modified scratch storage package as dependency

taybenlor opened the pull request:
Just throwing this in so I can refer to this package from scratch-iframe

taybenlor committed:
Add modified scratch storage package as dependency

taybenlor committed:
Change a behaviour that prevented grok intercepting saves. Remove top bar from GUI.

wryun commented on src/lib/project-saver-hoc.jsx:
```diff
@@ -138,14 +138,18 @@ const ProjectSaverHOC = function (WrappedComponent) {
-            if (this.props.isShowingSaveable && this.props.autoSaveTimeoutId === null) {
+            if (this.props.autoSaveTimeoutId === null) {
+            // Grok wants autosave to happen regardless of whether saving is enabled
+            // if (this.props.isShowingSaveable && this.props.autoSaveTimeoutId === null) {
```
Maybe some explanation here? What does it mean to save an unsaveable project?

taybenlor committed:
Fixes per code review. Blast the autosaving to prevent it disappearing on project change.

taybenlor commented:
@wryun I removed some of that "no saving" stuff because it turns out it's unnecessary now that I've hidden the menubar.

I added some other stuff because I noticed if I rapidly change projects then autosaving breaks.

Ready for review

wryun commented on package.json:
```diff
@@ -108,11 +109,11 @@
     "scratch-paint": "0.2.0-prerelease.20191217213717",
     "scratch-render": "0.1.0-prerelease.20191217212645",
-    "scratch-storage": "1.3.2",
+    "scratch-storage": "git://github.com/groklearning/scratch-storage.git#grok-release-17-01-2020",
```
hang on, did you just tag a release off an unmerged branch?

wryun commented on src/components/gui/gui.jsx:
```diff
@@ -199,7 +199,9 @@ const GUIComponent = props => {
                     />
                 ) : null}
-                <MenuBar
+                {/* Grok: We want the menubar hidden for our interface
```
not as advertised on the tin :)

I mean, it's nice-ish to do this, though we could hide it in the CSS in scratch-iframe instead if we're being lazy (and then we carry fewer random hacks in this repo)... choose your poison?

wryun reviewed (approved):
LGTM :ghost: (with minor caveats as mentioned)

taybenlor committed:
Update .npmignore

taybenlor committed:
Fix scratch storage dependency

taybenlor committed:
Fixes to autosaving behaviour

taybenlor committed:
Force a bonus autosave

wryun reviewed (approved):
Still LGTM (aside from earlier comments).
